### PR TITLE
feat: refactor WriterProperties class

### PR DESCRIPTION
**Pull request by ion-elgreco**
# Description
- Add better typehint
- Better description of the compression levels
- Gracefully handle wrong compression levels, rust parquet errors were not clear enough

### CODE REVIEW

**Review by rtyler — APPROVED** [review]

File: python/deltalake/table.py [hunk 1]
```diff
@@ -62,39 +62,70 @@ def __init__(
...
+            Literal[
+                "UNCOMPRESSED",
+                "SNAPPY",
+                "GZIP",
+                "BROTLI",
+                "LZ4",
+                "ZSTD",
+                "LZ4_RAW",
+            ]
+        ] = None,
```
Comment (roeap) [inline python/deltalake/table.py:75]: Do not have too strong opinions about this, just thinking :) - could / should this also be an enum?

**Review by roeap — APPROVED** [review]

File: python/tests/test_writerproperties.py [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+
+def test_write_with_writerproperties(
+    tmp_path: pathlib.Path, sample_table: pa.Table, writer_properties: WriterProperties
+):
+    write_deltalake(
+        tmp_path, sample_table, engine="rust", writer_properties=writer_properties
+    )
+
+    parquet_path = DeltaTable(tmp_path).file_uris()[0]
+    print(parquet_path)
```
Comment (roeap) [inline python/tests/test_writerproperties.py:86]: debug artifact?

File: python/deltalake/table.py [hunk 3]
```diff
@@ -51,6 +52,59 @@
...
+    UNCOMPRESSED = "UNCOMPRESSED"
+    SNAPPY = "SNAPPY"
+    GZIP = "GZIP"
+    BROTLI = "BROTLI"
+    LZ4 = "LZ4"
+    ZSTD = "ZSTD"
+    LZ4_RAW = "LZ4_RAW"
+
+    @classmethod
+    def __from_str__(cls, value: str) -> "Compression":
```
Comment (roeap) [inline python/deltalake/table.py:65]: is this an "official" dunder method? if not, could it be just `from_str`?

File: python/tests/test_writerproperties.py (hunk 2 above)
Comment (ion-elgreco) [reply python/tests/test_writerproperties.py:86]: Ah yup, good spot

File: python/deltalake/table.py (hunk 3 above)
Comment (ion-elgreco) [reply python/deltalake/table.py:65]: It's not, will change it

**Review by roeap — APPROVED** [review]

**Pull request merged**